txn_writer.go

```diff
@@ -288,17 +284,12 @@ func (payload SafeWriterPayload) validate(root string, sm gps.SourceManager) err
 // This mostly guarantees that dep cannot exit with a partial write that would
 // leave an undefined state on disk.
 func (sw *SafeWriter) Write(root string, sm gps.SourceManager, noExamples bool) error {
-
-	if sw.Payload == nil {
-		return errors.New("Cannot call SafeWriter.Write before SafeWriter.Prepare")
-	}
```

Comment:
My opinion is that I shouldn't have exported SafeWriter to begin with and that people other than us shouldn't be using it. 😀 I don't have it in front of me, but if we can just rename to `safeWriter` without running into issues, that would be good.